tests/integration/test_squeeze.py

```diff
@@ -13,39 +13,140 @@
+    (0,),
+    (1),
+    (DIM),
+    (0, 1),
+    (1, 0),
+    (1, 1),
+    (1, DIM),
+    (DIM, 1),
+    (DIM, DIM),
+    (1, 0, 0),
+    (1, 1, 0),
+    (1, 0, 1),
+    (1, 1, 1),
+    (DIM, 1, 1),
+    (1, DIM, 1),
+    (1, 1, DIM),
+    (DIM, DIM, DIM),
+]
+
+
+@pytest.mark.xfail
+def test_none_array_compare():
+    res_num = num.squeeze(None)  # AttributeError: 'NoneType'
+    res_np = np.squeeze(None)  # return None
+    assert np.array_equal(res_num, res_np, equal_nan=True)
+
+
+def test_none_array():
+    # numpy returned None
+    msg = r"NoneType"
+    with pytest.raises(AttributeError, match=msg):
+        num.squeeze(None)
+
+
+def test_num_invalid_axis():
+    size = (1, 2, 1)
+    a = num.random.randint(low=-10, high=10, size=size)
+    msg = r"one"
+    with pytest.raises(ValueError, match=msg):
+        num.squeeze(a, axis=1)
+
+
+def test_array_invalid_axis():
+    size = (1, 2, 1)
+    a = num.random.randint(low=-10, high=10, size=size)
+    msg = r"one"
+    with pytest.raises(ValueError, match=msg):
+        a.squeeze(axis=1)
+
+
+def test_num_axis_out_bound():
+    size = (1, 2, 1)
+    a = num.random.randint(low=-10, high=10, size=size)
+    msg = r"bounds"
+    with pytest.raises(np.AxisError, match=msg):
+        num.squeeze(a, axis=3)
+
+
+def test_array_axis_out_bound():
+    size = (1, 2, 1)
+    a = num.random.randint(-10, 10, size=size)
+    msg = r"bounds"
+    with pytest.raises(np.AxisError, match=msg):
+        a.squeeze(axis=3)
+
+
+@pytest.mark.parametrize("axes", (-1, -3))
+def test_num_axis_negative(axes):
+    size = (1, 2, 1)
+    a = np.random.randint(low=-10, high=10, size=size)
+    b = num.array(a)
+    res_np = np.squeeze(a, axis=axes)
+    res_num = num.squeeze(b, axis=axes)
+    assert np.array_equal(res_num, res_np)
+
+
+@pytest.mark.parametrize("axes", (-1, -3))
+def test_array_axis_negative(axes):
+    size = (1, 2, 1)
+    a = np.random.randint(low=-10, high=10, size=size)
+    b = num.array(a)
+    res_np = a.squeeze(axis=axes)
+    res_num = b.squeeze(axis=axes)
+    assert np.array_equal(res_num, res_np)
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_num_basic(size):
+    a = np.random.randint(low=-10, high=10, size=size)
+    b = num.array(a)
+    res_np = np.squeeze(a)
+    res_num = num.squeeze(b)
+    assert np.array_equal(res_num, res_np)
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_array_basic(size):
+    a = np.random.randint(low=-10, high=10, size=size)
+    b = num.array(a)
+    res_np = a.squeeze()
+    res_num = b.squeeze()
+    assert np.array_equal(res_num, res_np)
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_num_axis(size):
+    a = np.random.randint(low=-10, high=10, size=size)
+    b = num.array(a)
+    for xis in a.shape:
+        if xis == 1:
+            k = a.shape.index(xis)
+            res_np = np.squeeze(a, axis=k)
+            res_num = num.squeeze(b, axis=k)
+            assert np.array_equal(res_num, res_np)
```

Comment:
Help me understand the purpose of this loop. Since the test only does something at all in the case when a shape item is "1", and since it does the same thing every time, it seems like this would be equivalent:
```
    a = np.random.randint(low=-10, high=10, size=size)
    b = num.array(a)
    k = a.shape.index(1)

    res_np = np.squeeze(a, axis=k)
    res_num = num.squeeze(b, axis=k)

    assert np.array_equal(res_num, res_np)
```

If the test needs to be restricted to shapes that have at least one "1" somewhere, then that can be done in the parametrization:
```
@pytest.mark.parametrize("size", (s for s in SIZES if 1 in s), ids=str)
```
Or am I misunderstanding something about the test?

Reply:
`@pytest.mark.parametrize("size", (s for s in SIZES if 1 in s), ids=str)`  is good , it will filter the array whose shapes contains 1.
For the shape like `(DIM, 1, 1)`，`k = a.shape.index(1)`  will return value 1，but we can input 1 and 2 as the value of axis. Although the results are the same, the values passed in by axis are different. I think it is necessary to distinguish.